gateway/src/compression/mod.rs

```diff
@@ -1,10 +1,79 @@
+//! todo
+
+#![allow(unused)] // todo
```

Comment:
I need to figure out if some of the functions here should *actually* be dead code now